File: examples/numexpr-edsl/julia/ffi-jl/Data_NumExpr_foreign.jl
# FFI for Data.NumExpr (core) — the reference interpreter's transcendental
# primitives. purejl resolves Data.Number.sin/pow via its builtin shim, but the
# core module declares its OWN math foreigns so the same source compiles on the
# JS and BEAM backends too; this shim backs them on purejl with Base.*.
numPow(x) = y -> Base.Float64(x) ^ Base.Float64(y)
numSin(x) = Base.sin(x)
numCos(x) = Base.cos(x)
numExp(x) = Base.exp(x)
numLog(x) = Base.log(x)
numSqrt(x) = Base.sqrt(x)

File: examples/numexpr-edsl/python/ffi-py/Data_NumExpr_foreign.py
# FFI for Data.NumExpr (numexpr-core) - the six irreducible machine-math
# primitives behind core's uniform interface (see core/src/Data/NumExpr.purs:
# one line per backend; this is the purepy line).
import math as _math

def numPow(x): return lambda y: x ** y
def numSin(x): return _math.sin(x)
def numCos(x): return _math.cos(x)
def numExp(x): return _math.exp(x)
def numLog(x): return _math.log(x)
def numSqrt(x): return _math.sqrt(x)
